core/src/main/scala/kafka/admin/ConsumerGroupCommand.scala

```diff
@@ -126,7 +134,22 @@ object ConsumerGroupCommand extends Logging {
     private var consumer: KafkaConsumer[String, String] = _
 
     def listGroups(): List[String] = {
-      adminClient.listAllConsumerGroupsFlattened().map(_.groupId)
+      val result = adminClient.listConsumerGroups(
+        withTimeoutMs(new ListConsumerGroupsOptions()))
+
+      try {
+        val listings = result.all().get().asScala
+        listings.map(_.groupId()).toList
+      }
+      catch {
+        case e : ExecutionException if e.getCause.isInstanceOf[KafkaException] =>
```

Comment:
It's a little unclear why we have the custom exception handling here. Previously we just raised all errors? Should we raise the cause in all cases?

Reply:
The way the new KafkaAdminClient works is a little bit different than previously. I have seen that certain kind of exceptions (e.g. SaslAuthenticationException) are now nested, and I wanted to keep similar behaviour.